src/gfn/containers/transitions.py

```diff
@@ -67,31 +67,35 @@ def __init__(
         self.env = env
         self.is_backward = is_backward
 
-        # Assert that all tensors are in the same device as the environment.
+        # Assert that all tensors are on the same device as the environment.
         device = self.env.device
         for obj in [states, actions, next_states]:
             assert obj.tensor.device == device if obj is not None else True
-        for tensor in [conditioning, is_done, log_rewards, log_probs]:
+        for tensor in [conditioning, is_terminating, log_rewards, log_probs]:
             assert tensor.device == device if tensor is not None else True
 
         self.states = states if states is not None else env.states_from_batch_shape((0,))
         assert len(self.states.batch_shape) == 1
 
         self.conditioning = conditioning
+        assert (
+            self.conditioning is None
+            or self.conditioning.shape == self.states.tensor.shape
+        )
 
         self.actions = (
             actions if actions is not None else env.actions_from_batch_shape((0,))
         )
         assert self.actions.batch_shape == self.states.batch_shape
 
-        self.is_done = (
-            is_done
-            if is_done is not None
+        self.is_terminating = (
+            is_terminating
+            if is_terminating is not None
             else torch.full(size=(0,), fill_value=False, dtype=torch.bool, device=device)
         )
         assert (
-            self.is_done.shape == (self.n_transitions,)
-            and self.is_done.dtype == torch.bool
+            self.is_terminating.shape == (self.n_transitions,)
```

Comment:
I really would prefer we don't introduce changes in semantics like this. It's not required. Can you change it back please.